### ISSUE TITLE
Test after training leads to 0.0 dice

### ISSUE THREAD

**Issue by pervinh** [Tencent/MedicalNet#25]
I trained my model for 2000 epochs. Loss value decreased with every batch. However when i use test.py with the same training data, i obtain 0.0 dice.

```
mean dice for class-1 is 0.0                                                                                                                                                 mean dice for class-2 is 0.0
```
What could be the reason for this?

**Comment by Dylanself123** [comment]
i found the pre-processing step may cause this problem

**Comment by pervinh** [comment]
@Dylanself123 
is it the normalization step in brain18.py?

**Comment by Yufengevan** [comment]
Hi,Have you resolve this problem? I have the same problem.

**Comment by Dylanself123** [comment]
> @Dylanself123
> is it the normalization step in brain18.py?

The crop step, leading the input images different between training and test. U can check the training_data_process and testing_data_process

**Comment by nuist-xinyu** [comment]
> > @Dylanself123
> > is it the normalization step in brain18.py?
> 
> The crop step, leading the input images different between training and test. U can check the training_data_process and testing_data_process

Did you solve the problem?
I have the same problem here, very confused.

**Comment by infinite-tao** [comment]
Thanks!
     I tried skipping the data processing,the training process is normal, loss is always falling, but the test is very unstable (sometimes dice = 0.2, sometimes dice = 0). Maybe I have another problem here.


Well happy every day!




On 11/26/2019 14:07，nuist-xinyu<notifications@github.com> wrote：

@Dylanself123
is it the normalization step in brain18.py?

The crop step, leading the input images different between training and test. U can check the training_data_process and testing_data_process

Did you solve the problem?
I have the same problem here, very confused.

**Comment by nuist-xinyu** [comment]
> Thanks! I tried skipping the data processing,the training process is normal, loss is always falling, but the test is very unstable (sometimes dice = 0.2, sometimes dice = 0). Maybe I have another problem here. Well happy every day! On 11/26/2019 14:07，nuist-xinyu<notifications@github.com> wrote： @Dylanself123 is it the normalization step in brain18.py? The crop step, leading the input images different between training and test. U can check the training_data_process and testing_data_process Did you solve the problem? I have the same problem here, very confused. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.



> Thanks! I tried skipping the data processing,the training process is normal, loss is always falling, but the test is very unstable (sometimes dice = 0.2, sometimes dice = 0). Maybe I have another problem here. Well happy every day! On 11/26/2019 14:07，nuist-xinyu<notifications@github.com> wrote： @Dylanself123 is it the normalization step in brain18.py? The crop step, leading the input images different between training and test. U can check the training_data_process and testing_data_process Did you solve the problem? I have the same problem here, very confused. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

Hello, thank you for your reply. I have recently used 3D segmentation to use this model, There are two types of goals I segmented，but when I tested, there was only one label and no other. How can I solve this problem, this problem has troubled me for a long time

Well happy every day!

**Comment by wangm-ting** [comment]
> Did you solve the problem?
> I have the same problem here, very confused.

when I use my own dataset as the test ,the dice=0 .Do you kown how to solve it？

**Comment by lander1003** [comment]
> > Did you solve the problem?
> > I have the same problem here, very confused.
> 
> when I use my own dataset as the test ,the dice=0 .Do you kown how to solve it？

Do you solve it now?

**Comment by lander1003** [comment]
> Thanks! I tried skipping the data processing,the training process is normal, loss is always falling, but the test is very unstable (sometimes dice = 0.2, sometimes dice = 0). Maybe I have another problem here. Well happy every day! On 11/26/2019 14:07，nuist-xinyu<notifications@github.com> wrote： @Dylanself123 is it the normalization step in brain18.py? The crop step, leading the input images different between training and test. U can check the training_data_process and testing_data_process Did you solve the problem? I have the same problem here, very confused. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

Did you solve it ?
